libs/hiraya/src/components/Button/index.tsx

```diff
@@ -1,12 +1,27 @@
-import { Button } from '@mui/material';
+import { ButtonProps, ButtonStyled } from './style';
+import FileDownloadIcon from '@mui/icons-material/FileDownload';
 
-interface ButtonProps {
-  label: string;
-  variant?: 'text' | 'outlined' | 'contained';
-}
-
-const ButtonUsage: React.FC<ButtonProps> = ({ ...props }) => {
-  return <Button variant={props.variant}>{props.label}</Button>;
+const Button: React.FC<ButtonProps> = ({ ...props }) => {
+  if (props.buttonWidthSizes === 'iconOnly') {
+    return (
+      <ButtonStyled
+        variants={props.variants}
+        buttonHeightSizes={props.buttonHeightSizes}
+        buttonWidthSizes={props.buttonWidthSizes}
+      >
+        <FileDownloadIcon />
+      </ButtonStyled>
+    );
+  }
+  return (
+    <ButtonStyled
+      variants={props.variants}
+      buttonHeightSizes={props.buttonHeightSizes}
+      buttonWidthSizes={props.buttonWidthSizes}
+    >
+      {props.label}
+    </ButtonStyled>
```

Comment:
Great job on this, Now we can make this simpler especially that we are passing the same props by doing it like this

```tsx
    <ButtonStyled
      variants={props.variants}
      buttonHeightSizes={props.buttonHeightSizes}
      buttonWidthSizes={props.buttonWidthSizes}
    >
      {buttonWidthSizes === "iconOnly" ? <FileDownloadIcon /> : props.label}
    </ButtonStyled>
```

Note that the icon in this case should be dynamic and come from the consumer, and not defined at the UI level.